aws-api/src/test/java/com/amplifyframework/api/aws/auth/OwnerBasedAuthTest.java

```diff
@@ -0,0 +1,351 @@
+        // Set API to use Cognito User Pools auth mode
+        apiName = GRAPHQL_API_WITH_COGNITO;
+
+        GraphQLRequest<OwnerAuth> request = ModelSubscription.onCreate(OwnerAuth.class);
+        GraphQLOperation<OwnerAuth> operation = subscribe(request);
+
+        assertNotNull(operation);
+        JSONAssert.assertEquals(Resources.readAsString("request-owner-auth.json"),
+                operation.getRequest().getContent(),
+                true);
+    }
+
+    /**
+     * Verify that owner argument is required for all subscriptions if ModelOperation.READ is specified
+     * while using Cognito User Pools auth mode.
+     * @throws AmplifyException if a ModelSchema can't be derived from the Model class.
+     */
+    @Test
+    public void ownerArgumentAddedForRestrictedReadWithUserPools() throws AmplifyException {
+        // Set API to use Cognito User Pools auth mode
+        apiName = GRAPHQL_API_WITH_COGNITO;
+
+        assertTrue(isOwnerArgumentAdded(Owner.class, SubscriptionType.ON_UPDATE));
+        assertTrue(isOwnerArgumentAdded(OwnerRead.class, SubscriptionType.ON_UPDATE));
+
+        assertTrue(isOwnerArgumentAdded(Owner.class, SubscriptionType.ON_DELETE));
+        assertTrue(isOwnerArgumentAdded(OwnerRead.class, SubscriptionType.ON_DELETE));
+
+        assertTrue(isOwnerArgumentAdded(Owner.class, SubscriptionType.ON_CREATE));
+        assertTrue(isOwnerArgumentAdded(OwnerRead.class, SubscriptionType.ON_CREATE));
+    }
+
+    /**
+     * Verify that owner argument is required for all subscriptions if ModelOperation.READ is specified
+     * while using OpenID Connect auth mode.
+     * @throws AmplifyException if a ModelSchema can't be derived from the Model class.
+     */
+    @Test
+    public void ownerArgumentAddedForRestrictedReadWithOidc() throws AmplifyException {
+        // Set API to use OpenID Connect auth mode
+        apiName = GRAPHQL_API_WITH_OIDC;
+
+        assertTrue(isOwnerArgumentAdded(OwnerOidc.class, SubscriptionType.ON_UPDATE));
+        assertTrue(isOwnerArgumentAdded(OwnerOidc.class, SubscriptionType.ON_DELETE));
+        assertTrue(isOwnerArgumentAdded(OwnerOidc.class, SubscriptionType.ON_CREATE));
+    }
+
+    /**
+     * Verify owner argument is NOT required if the subscription type is not one of the restricted operations.
+     * @throws AmplifyException if a ModelSchema can't be derived from the Model class.
+     */
+    @Test
+    public void ownerArgumentNotAddedIfOperationNotRestrictedWithUserPools() throws AmplifyException {
+        // Set API to use Cognito User Pools auth mode
+        apiName = GRAPHQL_API_WITH_COGNITO;
+
+        assertFalse(isOwnerArgumentAdded(OwnerCreate.class, SubscriptionType.ON_UPDATE));
+        assertFalse(isOwnerArgumentAdded(OwnerUpdate.class, SubscriptionType.ON_UPDATE));
+        assertFalse(isOwnerArgumentAdded(OwnerDelete.class, SubscriptionType.ON_UPDATE));
+
+        assertFalse(isOwnerArgumentAdded(OwnerCreate.class, SubscriptionType.ON_DELETE));
```

Comment:
I think you need some helper methods to simplify all of these imperative checks.

Maybe varargs:
```java
assertOwnerArguments(SubscriptionType.ON_DELETE,
    OwnerCreate.class, OwnerUpdate.class, OwnerDelete.class));
```
or:
```java
assertOwnerArguments(OwnerUpdate.class, SubscriptionType.values());
```
^ Then you could use a single `for`-loop to go over a list of `OnwerUpdate.class`, `OwnerDelete.class`, etc.?

Reply:
I wanted to maintain the original syntax on how the tests were originally written, but I may revisit it another time to create static methods for assertion.